coral/src/domain/schema-request/schema-request-api.ts

```diff
@@ -1,20 +1,29 @@
 import { KlawApiRequest, KlawApiResponse } from "types/utils";
 import api from "src/services/api";
 
-const createSchemaRequest = (
+async function createSchemaRequest(
   params: KlawApiRequest<"schemaUpload">
-): Promise<KlawApiResponse<"schemaUpload">> => {
+): Promise<KlawApiResponse<"schemaUpload">> {
   const schemaUploadParams = {
     ...params,
     // schemaversion and appname
     // should be hard coded at the moment
     schemaversion: "1.0",
     appname: "App",
   };
-  return api.post<
-    KlawApiResponse<"schemaUpload">,
-    KlawApiRequest<"schemaUpload">
-  >("/uploadSchema", schemaUploadParams);
-};
+  return new Promise((resolve, reject) => {
```

Comment:
If this type of errors hold true for all endpoints, we could add this logic into [api.ts](https://github.com/aiven/klaw/blob/main/coral/src/services/api.ts).

Reply:
I thought about that, but I thought if we add it in every API call (more tedious of course), then backend could change error handling endpoint by endpoint 🤔  So e.g. if they change it for `uploadSchema`, we remove it in here & are good, while the other endpoint still return the old errors. 

(Though I would add a utils function for that, so it's only adding/removing one function)

WDYT?

Reply:
I think a super strictly typed generic handler would be the most idiot proof solution. I assume the response you are handling here is pretty generic for also other endpoints. The implementation will stay pretty clean as we only one more step in the promise chain. I think the `api.ts` should abstract all "quirks" from the API implementation into a unified way.

Reply:
Will update the POC 🏃‍♀️  (may come back to you about typing tho :D )